### PR TITLE
InstructorHelp: Remove horizontal scroll for mobile devices #7467

### LINKED ISSUE #7467: InstructorHelp: Remove horizontal scroll for mobile devices
Facing a horizontal scroll in Instructor Help page for mobile devices. _[screenshot]_
screenshot from 2017-06-08 05-20-51

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #7467 

**Outline of Solution**

- Removed horizontal scroll that we were facing on mobile devices.
- Page is made responsive to devices > 330px

This PR would probably help us in solving #5616

### CODE REVIEW

File: src/main/webapp/partials/instructorHelpSessions.jsp [hunk 1]
```diff
@@ -7751,4 +7751,4 @@
     <div class="separate-content-holder">
         <hr>
     </div>
-</div>
+</div>
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpSessions.jsp]: This is causing the build failure. It's also an unrelated change.

**Comment by amarlearning** [comment]
Thanks for that @whipermr5. Ready for the review!

**Comment by amarlearning** [comment]
@whipermr5 I think that is not related to this issue and we should open a separate PR for that one. I open this PR for this issue found in this [PR#7113](https://github.com/TEAMMATES/teammates/pull/7113)

**Comment by whipermr5** [comment]
@amarlearning The root cause of the issue is that the UI elements used in the actual pages are not properly responsive. Since the instructor help page is supposed to be an exact mirror of what the instructor should see in the actual pages, both need to be changed. I feel that the scope of the current issue and PR can be widened to include this.

File: src/main/webapp/WEB-INF/tags/instructor/course/courseEditAccessControlEditDivForInstr.tag [hunk 2]
```diff
@@ -8,10 +8,8 @@
 <div id="accessControlEditDivForInstr${instructorPanel.index}">
     <c:if test="${instructorPanel.accessControlDisplayed}">
         <div class="form-group">
-            <div class="col-sm-3">
-                <label class="control-label pull-right">Access-level</label>
-            </div>
-
+            <label class="col-sm-3 control-label">Access-level:</label>
+
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/course/courseEditAccessControlEditDivForInstr.tag]: This is causing build failure.

**Comment by amarlearning** [comment]
@whipermr5 ready for review!

File: src/main/webapp/partials/instructorHelpAddEditInstructors.jsp [hunk 3]
```diff
@@ -213,7 +211,6 @@
                         <div class="panel-heading">
                             <strong>Instructor 1:</strong>
                             <div class="pull-right">
-
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpAddEditInstructors.jsp]: Unrelated change.

File: src/main/webapp/partials/instructorHelpGettingStarted.jsp [hunk 4]
```diff
@@ -313,10 +315,9 @@
                 </div>
                 <br>
                 <div class="helpSectionContent">
-
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpGettingStarted.jsp]: Unrelated change.

File: src/main/webapp/partials/instructorHelpGettingStarted.jsp [hunk 5]
```diff
@@ -345,11 +346,10 @@
 
                             <br>
                             <br>See
-                            <a href="#sessionTypes">here</a> for more info about session types.</li>
-
+                            <a href="#sessionTypes">here</a> for more info about session types.
+                        </li>
                     </ul>
                     <br>
-
```
Comment (whipermr5) [inline src/main/webapp/partials/instructorHelpGettingStarted.jsp]: Unrelated change.

File: src/main/webapp/WEB-INF/tags/instructor/course/courseEditAccessControlEditDivForInstr.tag [hunk 6]
```diff
@@ -8,10 +8,7 @@
 <div id="accessControlEditDivForInstr${instructorPanel.index}">
     <c:if test="${instructorPanel.accessControlDisplayed}">
         <div class="form-group">
-            <div class="col-sm-3">
-                <label class="control-label pull-right">Access-level</label>
-            </div>
-
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/course/courseEditAccessControlEditDivForInstr.tag]: This removal is also unrelated - there can be a blank line; just make sure there are no trailing spaces.

File: src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html [hunk 7]
```diff
@@ -1,219 +1,20 @@
...
-              </tbody>
-            </table>
-          </div>
-        </div>
-      </div>
+  <div class="row">
+    <div class="alert alert-warning col-md-4 col-md-offset-4">
+      <img src="/images/error.png" style="float: left; margin: 0 10px 10px 0; height: 90px;">
+      <p>
+        There was an error in our server.
```
Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: Marking this.

File: src/test/resources/pages/instructorSearchPageSearchCommentsAsHelper.html [hunk 8]
```diff
@@ -1,151 +1,20 @@
...
-              </tbody>
-            </table>
-          </div>
-        </div>
-      </div>
+  <div class="row">
+    <div class="alert alert-warning col-md-4 col-md-offset-4">
+      <img src="/images/error.png" style="float: left; margin: 0 10px 10px 0; height: 90px;">
+      <p>
+        There was an error in our server.
```
Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchCommentsAsHelper.html]: You need to do some sanity checking on GodMode changes before committing them wholesale.

**Comment by amarlearning** [comment]
Ready for review

File: src/test/resources/pages/newlyJoinedInstructorFeedbackSessionsPage.html [hunk 9]
```diff
@@ -0,0 +1,1215 @@
+<div class="container" id="mainContent">
```
Comment (whipermr5) [inline src/test/resources/pages/newlyJoinedInstructorFeedbackSessionsPage.html]: I believe this file was removed in #7511 and is no longer used.

**Comment by whipermr5** [comment]
This is a huge task making all the elements more responsive - I'm glad you chose to take it up. Here are some things to improve:

screen shot 2017-07-12 at 2 42 11 pm
screen shot 2017-07-12 at 2 42 44 pm
screen shot 2017-07-12 at 2 43 13 pm
screen shot 2017-07-12 at 2 43 54 pm
screen shot 2017-07-12 at 2 44 47 pm
screen shot 2017-07-12 at 2 45 06 pm
screen shot 2017-07-12 at 2 45 16 pm
screen shot 2017-07-12 at 2 46 04 pm
screen shot 2017-07-12 at 2 47 04 pm

Up to you whether you want to fix them all in this PR, or break them up into multiple PRs.

**Comment by whipermr5** [comment]
This is a useful tool for testing on multiple resolutions: http://quirktools.com/screenfly/

**Comment by amarlearning** [comment]
There is no problem in solving all the mentioned issues in this PR only. But if we can divide 2 element issue per PR then it would be more easy to work on that.

**Comment by amarlearning** [comment]
@whipermr5 This is already fixed. _[Old screenshot]_
Old screenshot

_[New Screenshot]_
screenshot from 2017-07-12 13-24-58

**Comment by whipermr5** [comment]
> Was this the screenshot from InstructorHelp page?

Yup I reviewed from instructor help since it's supposed to be a mirror of all the other UI elements.

**Comment by amarlearning** [comment]
As you know that all the elements in the InstructorHelp page are inside the container while on the other hand that's not the case with same elements present on the different pages. That's why there is some glitch seen.

**Comment by whipermr5** [comment]
Oh I see; didn't know that 😅 I guess it's fine then.

**Comment by amarlearning** [comment]
@whipermr5 What change do you want here _[screenshot]_?  And sorry for the delay :sweat_smile:

what_change_needed

**Comment by whipermr5** [comment]
@amarlearning The text is wrapping too early when there's still space before the edit button.

**Comment by amarlearning** [comment]
@whipermr5 let's finish this off :wink:

**Comment by amarlearning** [comment]
@whipermr5 reminder

**Comment by wkurniawan07** [comment]
Closing due to an inactivity for a very long period. If anyone is interested to continue the work done here, feel free to reopen/request for reopen.
